libcxx/include/__format/format_arg.h

```diff
@@ -144,7 +144,8 @@ _LIBCPP_HIDE_FROM_ABI decltype(auto) __visit_format_arg(_Visitor&& __vis, basic_
   __libcpp_unreachable();
 }
 
-#  if _LIBCPP_STD_VER >= 26
+#  if _LIBCPP_STD_VER >= 26 && (!defined(_LIBCPP_COMPILER_CLANG_BASED) || _LIBCPP_CLANG_VER >= 1800)
```

Comment:
https://clang.llvm.org/docs/ReleaseNotes.html

> Implemented P0847R7: Deducing this. Some related core issues were also implemented (CWG2553, CWG2554, CWG2653, CWG2687). Because the support for this feature is still experimental, the feature test macro __cpp_explicit_this_parameter was not set in this version.

I'll revert the change.

Reply:
Maybe something like:

```c++
#  if _LIBCPP_STD_VER >= 26 && (defined(__cpp_explicit_this_parameter) || !defined(_LIBCPP_COMPILER_CLANG_BASED) || _LIBCPP_CLANG_VER >= 1800)
...
#endif
```
or 

```c++
#if defined(__cpp_explicit_this_parameter) || !defined(_LIBCPP_COMPILER_CLANG_BASED) || _LIBCPP_CLANG_VER >= 1800
#define _LIBCPP_EXPLICIT_THIS_PARAMETER
#endif
```

Reply:
> https://clang.llvm.org/docs/ReleaseNotes.html
> 
> > Implemented P0847R7: Deducing this. Some related core issues were also implemented (CWG2553, CWG2554, CWG2653, CWG2687). Because the support for this feature is still experimental, the feature test macro __cpp_explicit_this_parameter was not set in this version.
> 
> I'll revert the change.

I looked at the status page https://clang.llvm.org/cxx_status.html which misses this detail. 

I would go with with a new config variable `_LIBCPP_HAS_EXPLICIT_THIS_PARAMETER`
```
// Clang-18 has support for deducing this, but it does not set the FTM.
#if defined(__cpp_explicit_this_parameter) || (defined(_LIBCPP_CLANG_VER ) &&_LIBCPP_CLANG_VER >= 1800))
#  define _LIBCPP_HAS_EXPLICIT_THIS_PARAMETER
#endif
```
This allows to easily add AppleClang when they support it. For the library we only support GCC and Clang based compilers.

Reply:
> I would go with with a new config variable `_LIBCPP_HAS_EXPLICIT_THIS_PARAMETER`
> 
> ```
> // Clang-18 has support for deducing this, but it does not set the FTM.
> #if defined(__cpp_explicit_this_parameter) || (defined(_LIBCPP_CLANG_VER ) &&_LIBCPP_CLANG_VER >= 1800))
> #  define _LIBCPP_HAS_EXPLICIT_THIS_PARAMETER
> #endif
> ```
> 
> This allows to easily add AppleClang when they support it. For the library we only support GCC and Clang based compilers.

Thank you! Done!

I think we also need to disable the test on Apple Clang, so I added:
`// UNSUPPORTED: clang-16 || clang-17 || apple-clang`

I hope this is the way.